### PR TITLE
Feature: Refactor NPID handling and introduce split functionality

### PR DESCRIPTION
**Pull request by krharum**
This pull request introduces significant changes to how NPIDs (Norwegian Personal Identification Numbers) are handled in the application, particularly focusing on splitting and merging operations. The changes streamline and generalize NPID-related commands and DTOs, introduce a new "split" operation for NPIDs, and update the order processing logic to accommodate these new flows. Additionally, there are improvements in code clarity and minor refactoring.

**NPID Handling Improvements:**

* The `PdlMergeNpidCommand` and `MergeIdent` classes have been renamed and generalized to `PdlNpidCommand` and `NpidIdentDTO`, respectively, with field names updated for clarity (e.g., `npid` → `otherIdent`). [[1]](diffhunk://#diff-8ce3360a7f71cd40ee9e4791c3bffb6eb1c45c6b352306e334f643368d04a8d1L16-R16) [[2]](diffhunk://#diff-bef5342975ad52f7d4f21f10402f61976d26498463625ffac7e38ae84932a2dbL15-R17) [[3]](diffhunk://#diff-4f4fc77d24dd7acb836d97e68af728e499bb85dac1c0a3f1b068c83efb763975L7-R13) [[4]](diffhunk://#diff-dee3815cb7ffa6ba521b18e304158b1e9eac58e17cea862327599b7b706ef936L15-R14)
* The order request and processing logic now supports a new "split" operation for NPIDs (`split` field in `OrdreRequest`), and the service layer creates and processes these split orders as part of the NPID flow. [[1]](diffhunk://#diff-0d79d5796f3f928e85bb567d034b07d21b875d9e5b084253635d81a012f2705bR17) [[2]](diffhunk://#diff-dee3815cb7ffa6ba521b18e304158b1e9eac58e17cea862327599b7b706ef936R238-R265) [[3]](diffhunk://#diff-4f4fc77d24dd7acb836d97e68af728e499bb85dac1c0a3f1b068c83efb763975L76-L83)
* The command dispatch logic in `PdlTestdataConsumer` has been updated to use the new `PdlNpidCommand` for both split and merge operations, and to use the new DTO structure. [[1]](diffhunk://#diff-4f4fc77d24dd7acb836d97e68af728e499bb85dac1c0a3f1b068c83efb763975R152-R158) [[2]](diffhunk://#diff-4f4fc77d24dd7acb836d97e68af728e499bb85dac1c0a3f1b068c83efb763975L167-R178)

**Order Processing and Filtering Enhancements:**

* The service logic for sending all information elements (`sendAlleInformasjonselementer`) now filters and processes NPID-related requests more precisely, ensuring correct handling of test and NPID identities, and improves filtering of alias creation to avoid duplicates.
* Helper methods have been added to `OpprettRequest` to check if an ident is an NPID, improving code readability and maintainability.

**Technical and Refactoring Updates:**

* Improved usage of Jackson's `JsonNode` for request bodies, replacing raw string serialization, which enhances type safety and reduces serialization errors. [[1]](diffhunk://#diff-4f4fc77d24dd7acb836d97e68af728e499bb85dac1c0a3f1b068c83efb763975L120-R127) [[2]](diffhunk://#diff-ca1272ef5a35230a20bc2e6d572abd81795e93ec83c27461b4ad9a57cf3673aeR15) [[3]](diffhunk://#diff-ca1272ef5a35230a20bc2e6d572abd81795e93ec83c27461b4ad9a57cf3673aeL26-R32)
* Minor code cleanups, such as removing unused imports, using streams and collectors for set operations, and updating variable declarations for clarity. [[1]](diffhunk://#diff-dee3815cb7ffa6ba521b18e304158b1e9eac58e17cea862327599b7b706ef936L5) [[2]](diffhunk://#diff-dee3815cb7ffa6ba521b18e304158b1e9eac58e17cea862327599b7b706ef936L52-L57) [[3]](diffhunk://#diff-dee3815cb7ffa6ba521b18e304158b1e9eac58e17cea862327599b7b706ef936L211-R211) [[4]](diffhunk://#diff-ca1272ef5a35230a20bc2e6d572abd81795e93ec83c27461b4ad9a57cf3673aeL3)

These changes collectively improve the robustness, clarity, and extensibility of NPID-related operations in the system, preparing the codebase for future enhancements and reducing technical debt.

### CODE REVIEW

File: apps/pdl-forvalter/src/main/java/no/nav/pdl/forvalter/service/PdlOrdreService.java [hunk 1]
```diff
@@ -238,26 +235,34 @@ private Flux<OrdreResponseDTO.PdlStatusDTO> sendAlleInformasjonselementer(List<O
...
+                                .filter(OpprettRequest::isNpidIdent)
                                 .flatMap(this::npidMerge)
                                 .collectList(),
                         Flux.fromIterable(sorterteOpprettinger)
                                 .flatMap(oppretting -> aliasRepository.existsByPersonId(oppretting.getPerson().getId())
                                         .zipWith(Mono.just(oppretting)))
-                                .filter(exist -> isFalse(exist.getT1()))
+                                .filter(exist -> exist.getT2().isNotNpidIdent() ||
+                                                 isTrue(exist.getT1()))
                                 .flatMap(oppretting -> getOrdrer(oppretting.getT2()))
```
[the remark this answers is not in the document]
Comment (krharum) [reply apps/pdl-forvalter/src/main/java/no/nav/pdl/forvalter/service/PdlOrdreService.java:256]: Dette er ønsket oppførsel !

**Review by stigus — APPROVED** [review]
Nice 🤩
